core/src/main/scala/chisel3/Mem.scala

```diff
@@ -54,8 +55,8 @@ sealed abstract class MemBase[T <: Data](val t: T, val length: BigInt, sourceInf
     with SourceInfoDoc {
 
   if (t.isConst) Builder.error("Mem type cannot be const.")(sourceInfo)
-
-  requireNoProbeTypeModifier(t, "Cannot make a Mem of a Chisel type with a probe modifier.")(sourceInfo)
+  if (containsProbeTypeModifier(t))
+    Builder.error("Cannot make a Mem of a Chisel type with a probe modifier.")(sourceInfo)
```

Comment:
```suggestion
  if (containsProbeTypeModifier(t)) {
    Builder.error("Cannot make a Mem of a Chisel type with a probe modifier.")(sourceInfo)
  }
```
Style-nit but GOTO FAIL has scared me into requiring curly braces for if. Maybe we could enforce with scalafmt/scalafix.